### PR TITLE
[CARBONDATA-1444] Support Boolean data type

### PR DESCRIPTION
**Pull request by xubo245**
Spark/Hive table support Boolean data type, the internal table also should support Boolean data type.

   Boolean data type Range: TRUE or FALSE. Do not use quotation marks around the TRUE and FALSE literal values. You can write the literal values in uppercase, lowercase, or mixed case. The values queried from a table are always returned in lowercase, true or false.
    In implementing this function, we employ:

```
1.  endcoding: RLE
2.  data expression: byte array
```

        
    CarbonData should support boolean data type in following aspects:

```
1. create table: support Boolean data type
2. insert into table values: support insert Boolean column 
3. insert overwrite
4. insert into table select from another table
5. select from a table
6. load data: from a local csv file
7. filter: including >=, >, =, <=, <, =, !=, in, not in, and, or, is not null, is null, like
8. describle: should show boolean data type
```

We also add some test cases in booleantype directory of spark2:

```
1. BooleanDataTypesBaseTest: add some test cases for testing create table, describe table, altering table and add column, Altering table and drop column, insert values, and deleting table.
2. BooleanDataTypesBigFileTest: add some test cases for testing create table, insert into, filter table, sort_column, and unsafe module with big file, you can change the file size by changing trueNum value
3. BooleanDataTypesFilterTest:  add some test cases for testing filter function, including >=, >, =, <=, <, =, !=, in, not in, and, or, is not null, is null, and like. We also test it when loading twice
4. BooleanDataTypesInsertTest: add some test cases for testing insert into table values, insert into table select from another table, insert overwrite, inserting into Hive table from carbon table, inserting into carbon table from Hive table, inserting with the number of data type in source and target table columns being different, inserting with the order of data type in source and target table columns being different
5. BooleanDataTypesLoadTest: add some test cases for testing load table, including one boolean column, boolean column and other data type, load with different parameters, create table with different parameters and then load, bad_records_action, load overwrite, and unsafe module.
6. BooleanDataTypesSortTest: add some test cases for testing sort_columns is boolean column, sort_columns  is two boolean columns and other data type column
7. BooleanDataTypesParameterTest:  add some test cases for testing ENABLE_AUTO_LOAD_MERGE is false or true when create table, load table, insert table, sort_column
```

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/75/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/830/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/198/

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 1]
```diff
@@ -707,6 +711,10 @@
    */
   public static final char BIG_INT_MEASURE = 'd';
   /**
+   * BOOLEAN_VALUE_MEASURE
+   */
+  public static final char BOOLEAN_MEASURE = 'a';
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: This is not required. Others are for backward compatible only.

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 2]
```diff
@@ -23,6 +23,10 @@
 
 public final class CarbonCommonConstants {
   /**
+   * Boolean size in bytes
+   */
+  public static final int BOOLEAN_SIZE_IN_BYTE = 1;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: This is not required, it should be added in DataType enum

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java [hunk 3]
```diff
@@ -187,6 +179,7 @@ public static ColumnPage newPage(TableSpec.ColumnSpec columnSpec, DataType dataT
         case BYTE:
         case SHORT:
         case SHORT_INT:
+        case BOOLEAN:
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: Move it before BYTE. Please modify the same in all switch case

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java [hunk 4]
```diff
@@ -271,6 +267,18 @@ private static ColumnPage newIntPage(TableSpec.ColumnSpec columnSpec, int[] intD
     return columnPage;
   }
 
+  private static ColumnPage newBooleanPage(TableSpec.ColumnSpec columnSpec, byte[] booleanData) {
+    ColumnPage columnPage = createPage(columnSpec, BOOLEAN, booleanData.length);
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: To make it simple, I think it is better to store boolean as byte in ColumnPage, so you can create a ColumnPage of Byte type.
When come to encoding, you still can do proper encoding of this column page.

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java [hunk 5]
```diff
@@ -92,6 +90,10 @@ public ColumnPageDecoder createDecoder(List<Encoding> encodings, List<ByteBuffer
       RLEEncoderMeta metadata = new RLEEncoderMeta();
       metadata.readFields(in);
       return new RLECodec().createDecoder(metadata);
+    } else if (encoding == BOOL_BYTE) {
+      BooleanEncoderMeta metadata=new BooleanEncoderMeta();
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java]: Not compliant to coding standard. Add space before and after `=`

Comment (jackylk) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java]: I think it is better to use RLE on byte, since it just has two values

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.datastore.page.encoding.bool;
+
+/**
+ * convert tools for boolean data type
+ */
+public class BooleanConvert {
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java]: I think this is not required, it is simple conversion only.

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanEncoderMeta.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+public class BooleanEncoderMeta extends ColumnPageEncoderMeta implements Writable {
+  private String compressorName;
+
+  public BooleanEncoderMeta() {
+  }
+
+  public BooleanEncoderMeta(TableSpec.ColumnSpec columnSpec, DataType storeDataType,
+                            SimpleStatsResult stats, String compressorName) {
+    super(columnSpec,storeDataType,stats,compressorName);
+    this.compressorName = compressorName;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanEncoderMeta.java]: compressorName is already stored in ColumnPageEncoderMeta, this BooleanEncoderMeta is not required.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/278/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/154/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/910/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/279/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/155/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/911/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/280/

**Comment by xubo245** [comment]
I changed the format module.Other module can not invoke the new code in format module. 
Could you help me to deal with it? @ravipesala

**Comment by ravipesala** [comment]
@xubo245 Please use `mvn clean -Pbuild-with-format install` to build fomat module

**Comment by xubo245** [comment]
I can run success in my computer, but I can not fix this problem in github/jenkins. How to add "mvn clean -Pbuild-with-format install" to run my branch in github after pushing request to github?  @ravipesala

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 2 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: ok

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/325/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/201/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/953/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/328/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/204/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/956/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/205/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/957/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/329/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/207/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/331/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/959/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/210/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/962/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/334/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/211/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/963/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/335/

**Comment by xubo245** [comment]
Please update jar of format module @ravipesala

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/336/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/212/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/213/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/337/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/964/

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 1 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: ok, I have removed it.

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/965/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/214/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/338/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/215/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/339/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/966/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/967/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/216/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/340/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/968/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/217/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/341/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/969/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/260/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1013/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/382/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/384/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/262/

**Comment by xubo245** [comment]
@jackylk please review it

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1015/

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java [hunk 8]
```diff
@@ -206,6 +199,7 @@ public static ColumnPage newPage(TableSpec.ColumnSpec columnSpec, DataType dataT
       }
     } else {
       switch (dataType) {
+        case BOOLEAN:
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: It is better to add a `newBooleanPage`, otherwise `newBytePage` internal will store the data type as byte instead of boolean

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java [hunk 9]
```diff
@@ -436,6 +439,11 @@ public void putData(int rowId, Object value) {
   public abstract void putShortInt(int rowId, int value);
 
   /**
+   * Set boolean value at rowId
+   */
+  public abstract void putBoolean(int rowId, boolean value);
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: Since internally boolean is stored as byte, so you can implement this function to call `putByte` instead of abstract function

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java [hunk 10]
```diff
@@ -321,6 +315,11 @@ private static ColumnPage newLVBytesPage(TableSpec.ColumnSpec columnSpec,
   public abstract void setShortIntPage(byte[] shortIntData);
 
   /**
+   * Set boolean values to page
+   */
+  public abstract void setBooleanPage(byte[] booleanData);
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: Since internally boolean is stored as byte, so you can implement this function to call setBytePage instead of abstract function

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java [hunk 11]
```diff
@@ -485,6 +496,11 @@ private void putNull(int rowId) {
   public abstract int getShortInt(int rowId);
 
   /**
+   * Get boolean value at rowId
+   */
+  public abstract boolean getBoolean(int rowId);
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: The same as above, you can implement this function here instead of abstract

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java [hunk 12]
```diff
@@ -530,6 +546,11 @@ private void putNull(int rowId) {
   public abstract byte[] getShortIntPage();
 
   /**
+   * Get boolean value page
+   */
+  public abstract byte[] getBooleanPage();
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: The same as above, you can implement this function here instead of abstract

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java [hunk 13]
```diff
@@ -38,11 +38,7 @@
 import org.apache.carbondata.core.util.CarbonUtil;
 import org.apache.carbondata.format.Encoding;
 
-import static org.apache.carbondata.format.Encoding.ADAPTIVE_DELTA_INTEGRAL;
-import static org.apache.carbondata.format.Encoding.ADAPTIVE_FLOATING;
-import static org.apache.carbondata.format.Encoding.ADAPTIVE_INTEGRAL;
-import static org.apache.carbondata.format.Encoding.DIRECT_COMPRESS;
-import static org.apache.carbondata.format.Encoding.RLE_INTEGRAL;
+import static org.apache.carbondata.format.Encoding.*;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java]: Do not use wildcard

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java [hunk 14]
```diff
@@ -203,7 +203,7 @@ private void initMeasureBlockIndexes() {
       } else {
         // specific for restructure case where default values need to be filled
         pageNumbers = blockChunkHolder.getDataBlock().numberOfPages();
-        numberOfRows = new int[] { blockChunkHolder.getDataBlock().nodeSize() };
+        numberOfRows = new int[] { blockChunkHolder.getDataBlock().nodeSize()};
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java]: do not modify this

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java [hunk 15]
```diff
@@ -215,7 +215,7 @@ private void initMeasureBlockIndexes() {
...
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java]: do not modify this

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  /**
+   * convert byte to boolean
+   *
+   * @param data byte type data
+   * @return boolean type data
+   */
+  public static boolean byte2Boolean(int data) {
+    return data == 1 ? true : false;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java]: simply it

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+  public static final byte trueValue = 1;
+  public static final byte falseValue = 0;
+  /**
+   * convert boolean to byte
+   *
+   * @param data data of boolean data type
+   * @return byte type data by convert
+   */
+  public static byte boolean2Byte(boolean data) {
+    return data ? (byte) 1 : (byte) 0;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java]: use constant

File: processing/src/main/java/org/apache/carbondata/processing/store/TablePage.java [hunk 18]
```diff
@@ -187,6 +188,12 @@ private void convertToColumnarAndAddToPages(int rowId, CarbonRow row, byte[] mdk
           value != null) {
         value = ((Decimal) value).toJavaBigDecimal();
       }
+      if (measurePages[i].getColumnSpec().getSchemaDataType()
+              == DataType.BOOLEAN && value != null) {
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/store/TablePage.java]: incorrect line wrap

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java [hunk 19]
```diff
@@ -361,6 +360,10 @@ public void putData(int rowId, Object value) {
       return;
     }
     switch (dataType) {
+      case BOOLEAN:
+        putByte(rowId, (byte) value);
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: use `putBoolean`

File: integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/BooleanDataTypesLoadTest.scala [hunk 20]
```diff
@@ -0,0 +1,655 @@
...
+import org.apache.spark.sql.test.util.QueryTest
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+
+/**
+  * Created by root on 9/17/17.
+  */
+class BooleanDataTypesLoadTest extends QueryTest with BeforeAndAfterEach with BeforeAndAfterAll {
+  val rootPath = new File(this.getClass.getResource("/").getPath
+    + "../../../..").getCanonicalPath
+
```
Comment (jackylk) [inline integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/BooleanDataTypesLoadTest.scala]: add some testcase to use unsafe while loading

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/277/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/399/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1030/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/402/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/279/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1033/

**Comment by xubo245** [comment]
@jackylk  CI pass

**Comment by xubo245** [comment]
@jackylk  please review it

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java (hunk 5 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java]: ok, I have changed the encoding from snappy to RLE

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java (hunk 6 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java]: I think it should be reserved. There are some place invoke it.

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java (hunk 3 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: ok

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java (hunk 4 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: ok, I use newBytePage to replace newBooleanPage

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanEncoderMeta.java (hunk 7 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanEncoderMeta.java]: ok

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java (hunk 9 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: ok, I have changed it

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java (hunk 10 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: ok, I have changed it

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java (hunk 11 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: ok, I have changed it

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java (hunk 10 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: ok, I have removed it

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java (hunk 12 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: Ok, I have changed it.

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java (hunk 13 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java]: Ok, I have changed it.

Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/EncodingFactory.java]: I also changed similar problem in columnPage

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java (hunk 14 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java]: Ok.

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java (hunk 15 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelFilterExecuterImpl.java]: Ok.

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java (hunk 16 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java]: Ok.

File: core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java (hunk 17 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/bool/BooleanConvert.java]: Ok.

File: processing/src/main/java/org/apache/carbondata/processing/store/TablePage.java (hunk 18 above)
Comment (xubo245) [reply processing/src/main/java/org/apache/carbondata/processing/store/TablePage.java]: Ok, I have changed it.

File: core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java (hunk 19 above)
Comment (xubo245) [reply core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java]: I have deleted it.

File: integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/BooleanDataTypesLoadTest.scala (hunk 20 above)
Comment (xubo245) [reply integration/spark2/src/test/scala/org/apache/carbondata/spark/testsuite/booleantype/BooleanDataTypesLoadTest.scala]: OK, I add  some testcase to use unsafe in BooleanDataTypesLoadTest.scala and BooleanDataTypesBigFileTest.scala

**Comment by xubo245** [comment]
@jackylk  I have added some descriptions for test cases in this PR.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/308/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1062/

**Comment by xubo245** [comment]
@jackylk The conflicts have been solved

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/454/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/329/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1082/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/455/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/330/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1083/

**Comment by xubo245** [comment]
Conflicts has been fixed， please review it @jackylk

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/368/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/493/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1121/

**Comment by xubo245** [comment]
@jackylk Please review it.

**Comment by jackylk** [comment]
LGTM

**Comment by jackylk** [comment]
Thanks for working on this
